src/main/java/org/apache/commons/lang3/Conversion.java

```diff
@@ -0,0 +1,1563 @@
+     * Converts a hexadecimal digit into binary (represented as boolean array) using the Msb0
+     * bit ordering.
+     * </p>
+     * <p>
+     * '1' is converted as follow: (0, 0, 0, 1)
+     * </p>
+     *
+     * @param hexDigit the hexadecimal digit to convert
+     * @return a boolean array with the binary representation of {@code hexDigit}
+     * @throws IllegalArgumentException if {@code hexDigit} is not a hexadecimal digit
+     */
+    public static boolean[] hexDigitMsb0ToBinary(final char hexDigit) {
+        switch (hexDigit) {
+        case '0':
+            return FFFF.clone();
+        case '1':
+            return FFFT.clone();
+        case '2':
+            return FFTF.clone();
+        case '3':
+            return FFTT.clone();
+        case '4':
+            return FTFF.clone();
+        case '5':
+            return FTFT.clone();
+        case '6':
+            return FTTF.clone();
+        case '7':
+            return FTTT.clone();
+        case '8':
+            return TFFF.clone();
+        case '9':
+            return TFFT.clone();
+        case 'a':// fall through
+        case 'A':
+            return TFTF.clone();
+        case 'b':// fall through
+        case 'B':
+            return TFTT.clone();
+        case 'c':// fall through
+        case 'C':
+            return TTFF.clone();
+        case 'd':// fall through
+        case 'D':
+            return TTFT.clone();
+        case 'e':// fall through
+        case 'E':
+            return TTTF.clone();
+        case 'f':// fall through
+        case 'F':
+            return TTTT.clone();
+        default:
+            throw new IllegalArgumentException("Cannot interpret '" + hexDigit + "' as a hexadecimal digit");
+        }
+    }
+
+    /**
+     * <p>
+     * Converts binary (represented as boolean array) to a hexadecimal digit using the default
+     * (Lsb0) bit ordering.
+     * </p>
+     * <p>
+     * (1, 0, 0, 0) is converted as follow: '1'
+     * </p>
+     *
+     * @param src the binary to convert
+     * @return a hexadecimal digit representing the selected bits
+     * @throws IllegalArgumentException if {@code src} is empty
+     * @throws NullPointerException if {@code src} is {@code null}
+     */
+    public static char binaryToHexDigit(final boolean[] src) {
+        return binaryToHexDigit(src, 0);
+    }
+
+    /**
+     * <p>
+     * Converts binary (represented as boolean array) to a hexadecimal digit using the default
+     * (Lsb0) bit ordering.
+     * </p>
+     * <p>
+     * (1, 0, 0, 0) is converted as follow: '1'
+     * </p>
+     *
+     * @param src the binary to convert
+     * @param srcPos the position of the lsb to start the conversion
+     * @return a hexadecimal digit representing the selected bits
+     * @throws IllegalArgumentException if {@code src} is empty
+     * @throws NullPointerException if {@code src} is {@code null}
+     */
+    public static char binaryToHexDigit(final boolean[] src, final int srcPos) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 20. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 313-324, lines 325-330 into separate methods.